blog/views.py

```diff
@@ -0,0 +1,44 @@
+from django.shortcuts import redirect
+from django.views.generic import ListView, DetailView
+
+from blog.forms import CommentaryForm
+from blog.models import Post
+
+
+# Create your views here.
+class PostListView(ListView):
+    model = Post
+    paginate_by = 5
+
+
+class PostDetailView(DetailView):
+    model = Post
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["form"] = (
+            kwargs.get("form") or CommentaryForm()
+        )  # Load the form or pass the one with errors
+        return context
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()  # Get the current post
+        form = CommentaryForm(request.POST)
+
+        # If the user is not authenticated,
+        # make the form invalid and add an error
+        if not request.user.is_authenticated:
```

Comment:
It's generally recommended to place authentication checks at the beginning of the function. Doing so can help to prevent unnecessary operations in case the user is not authenticated.